karaf-features-gen/src/main/groovy/com/github/lburgazzoli/gradle/plugin/KarafFeaturesGenTask.groovy

```diff
@@ -37,7 +39,20 @@ class KarafFeaturesGenTask extends DefaultTask {
     public static final String NAME = 'generateKarafFeatures'
 
     public KarafFeaturesGenTask() {
-        getOutputs().upToDateWhen(Specs.satisfyNone());
+        project.afterEvaluate {
+            extension.projects.each { selectedProject ->
+                // we need access the jar for any project we generate feature for
+                dependsOn( selectedProject.tasks.jar )
+
+                // we also want our inputs to be based on the runtime configuration
+                inputs.files( selectedProject.configurations.runtime )
+            }
+
+            // if there is an output file, add that as an output
+            if ( extension.outputFile != null ) {
+                outputs.file( extension.outputFile )
+            }
+        }
```

Comment:
Ultimately this should account for some representation of the extension values too.